source/extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.cc

```diff
@@ -0,0 +1,45 @@
+// NOLINT(namespace-envoy)
+//
+// This file is part of the QUICHE platform implementation, and is not to be
+// consumed or referenced directly by other Envoy code. It serves purely as a
+// porting layer for QUICHE.
+
+#include "extensions/quic_listeners/quiche/platform/quic_mem_slice_impl.h"
+
+#include "envoy/buffer/buffer.h"
+
+#include "common/common/assert.h"
+
+namespace quic {
+
+QuicMemSliceImpl::QuicMemSliceImpl(QuicBufferAllocator* /*allocator*/, size_t length) {
+  Envoy::Buffer::RawSlice iovec;
+  uint64_t num_iov = single_slice_buffer_.reserve(length, &iovec, 1);
+  ASSERT(num_iov == 1);
+  // evbuffer may return a slice longer than needed, trim it to requested length.
```

Comment:
nit: we are moving away from evbuffer internally. I would rephrase the comment to say this is how Envoy's buffer implementation works, etc.

Reply:
done